bundles/AdminBundle/Security/Guard/AdminAuthenticator.php

```diff
@@ -134,7 +134,10 @@ public function start(Request $request, AuthenticationException $authException =
             return $response;
         }
 
-        $event = new LoginRedirectEvent('pimcore_admin_login', ['perspective' => strip_tags($request->get('perspective'))]);
+        $perspective = $request->get('perspective');
+        $perspective = ($perspective !== null)? strip_tags($perspective): [];
+
+        $event = new LoginRedirectEvent('pimcore_admin_login', ['perspective' => $perspective ]);
```

Comment:
Shouldn't this be enough?

```suggestion
        $event = new LoginRedirectEvent('pimcore_admin_login', ['perspective' => strip_tags($request->get('perspective', ''))]);
```